packages/svelte/src/compiler/migrate/index.js

```diff
@@ -53,8 +53,17 @@ export function migrate(source) {
 			props_name: analysis.root.unique('props').name,
 			rest_props_name: analysis.root.unique('rest').name,
 			end: source.length,
-			run_name: analysis.root.unique('run').name,
-			needs_run: false
+			legacy_imports_names: new Map([
+				['run', analysis.root.unique('run').name],
+				['handlers', analysis.root.unique('handlers').name],
+				['stopImmediatePropagation', analysis.root.unique('stopImmediatePropagation').name],
+				['preventDefault', analysis.root.unique('preventDefault').name],
+				['stopPropagation', analysis.root.unique('stopPropagation').name],
+				['once', analysis.root.unique('once').name],
+				['self', analysis.root.unique('self').name],
+				['trusted', analysis.root.unique('trusted').name]
+			]),
```

Comment:
would make more sense for this to be an object — a `Map` is more appropriate when the shape isn't fixed (i.e. there isn't a finite set of keys known at author time) or the keys aren't strings, but that doesn't apply here

```suggestion
			legacy_imports_names: {
				run: analysis.root.unique('run').name,
				handlers: analysis.root.unique('handlers').name,
				stopImmediatePropagation: analysis.root.unique('stopImmediatePropagation').name,
				preventDefault: analysis.root.unique('preventDefault').name,
				stopPropagation: analysis.root.unique('stopPropagation').name,
				once: analysis.root.unique('once').name,
				self: analysis.root.unique('self').name,
				trusted: analysis.root.unique('trusted').name
			},
```